auth/set_project.go

```diff
@@ -2,15 +2,44 @@ package auth
 
 import (
 	"context"
+	"fmt"
 
+	management "github.com/ninech/apis/management/v1alpha1"
 	"github.com/ninech/nctl/api"
 	"github.com/ninech/nctl/api/config"
+	"github.com/ninech/nctl/internal/format"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
 )
 
 type SetProjectCmd struct {
 	Name string `arg:"" predictor:"resource_name" help:"Name of the default project to be used."`
 }
 
 func (s *SetProjectCmd) Run(ctx context.Context, client *api.Client) error {
-	return config.SetContextProject(client.KubeconfigPath, client.KubeconfigContext, s.Name)
+	org, err := client.Organization()
+	if err != nil {
+		return err
+	}
+
+	// we get the project without using the result to be sure it exists and the
+	// user has access.
+	if err := client.Get(ctx, types.NamespacedName{Name: s.Name, Namespace: org}, &management.Project{}); err != nil {
+		if !errors.IsNotFound(err) && !errors.IsForbidden(err) {
+			return err
+		}
+		if errors.IsNotFound(err) {
+			format.PrintWarningf("did not find Project %s in your Organization %s.\n", s.Name, org)
```

Comment:
We don't throw an error here - that's fine.

But I just wonder if we're not making things more confusing. For example: in this situation I can set a non-existent project "idonotexist". Then I can for example ask for releases in a separate `nctl get releases` query and I get: `no Releases found in project idontexist` (I could have forgotten that I had this warning that this project doesn't exist). What I'm trying to say is that it would be nice if we could have this "warning" somehow persistent throughout the nctl context. But that won't be so easy to change, I think.

A similar problem is with set org - you can just change the organization in your kubeconfig and `nctl auth whoami` will just read it without warning:

```
You are currently logged in the with the following account: "pawel@nine.ch"
Your current organization: "supersecretnonexistentorganization"

Available Organizations:
	nine

To switch the organization use the following command:
$ nctl auth set-org <org-name>
```

and `nctl get releases` will just report than no releases found (project/namespace name doesn't matter here):
```
no Releases found in project <projectname>
```

Reply:
One problem is projects are namespaced and live in the org namespace. So in it could be that a project exists in another organization so if we would throw an error, the user would first need to switch the org before being able to switch the project.

> What I'm trying to say is that it would be nice if we could have this "warning" somehow persistent throughout the nctl context. But that won't be so easy to change, I think.

Yeah I think that would be pretty hard to achieve. We could make a `Get` to the project before every other call we make but then we'd have a similar problem where you could be part of two organizations and you make a `nctl get app x -p bar` while having the org set to `foo`. This would then error even though we are also part of the `bar` org and are allowed to get that app.

Reply:
that's true. I guess we don't change anything now since your solution works. 
But maybe we'll come back to this discussion at some point in the future 🤷‍♂️ 😄

Reply:
Yeah I'm pretty sure it will be discussed again at some point :smile: